### ISSUE TITLE
_pytest.assertion.reinterpret.AssertionError: Test coverage should never be `None`

### ISSUE THREAD

**Issue by msabramo** [pytest-dev/pytest-cov#84]
With pytest-cov 2.0, I'm getting:

```
_pytest.assertion.reinterpret.AssertionError: Test coverage should never be `None`
```

On the same projects with pytest-cov 1.8.1, it works fine.

Example command-line:

```
$ tox -e py27
...
------------------------------------------------------------------------------------------------------------------------------- generated xml file: /home/monkey/ce402c011ae38dae/junit.xml --------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------- coverage: platform linux2, python 2.7.6-final-0 --------------------------------------------------------------------------------------------------------------------------------------
Name                                                    Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------------------
usersvc/__init__                                          196      0   100%
usersvc/appstatus                                          27     21    22%   9-16, 20-27, 31-38
usersvc/lib/__init__                                        2      0   100%
usersvc/lib/cache                                           6      0   100%
...
Traceback (most recent call last):
  File ".tox/py27/bin/py.test", line 11, in <module>
    sys.exit(main())
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/config.py", line 41, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 521, in __call__
    return self._docall(self.methods, kwargs)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 528, in _docall
    firstresult=self.firstresult).execute()
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 394, in execute
    res = method(*args)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/main.py", line 116, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/main.py", line 109, in wrap_session
    exitstatus=session.exitstatus)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 521, in __call__
    return self._docall(self.methods, kwargs)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 528, in _docall
    firstresult=self.firstresult).execute()
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 393, in execute
    return wrapped_call(method(*args), self.execute)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 109, in wrapped_call
    wrap_controller.send(call_outcome)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/terminal.py", line 362, in pytest_sessionfinish
    self.config.hook.pytest_terminal_summary(terminalreporter=self)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 521, in __call__
    return self._docall(self.methods, kwargs)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 528, in _docall
    firstresult=self.firstresult).execute()
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/_pytest/core.py", line 394, in execute
    res = method(*args)
  File "/home/monkey/ce402c011ae38dae/.tox/py27/local/lib/python2.7/site-packages/pytest_cov/plugin.py", line 164, in pytest_terminal_summary
    assert total is not None, 'Test coverage should never be `None`'
_pytest.assertion.reinterpret.AssertionError: Test coverage should never be `None`
ERROR: InvocationError: '/home/monkey/ce402c011ae38dae/.tox/py27/bin/py.test --teamcity --cov=usersvc --cov-report=html --cov-report=xml --cov-report=annotate'
_________________________________________________________________________________________________________________________________________________________ summary __________________________________________________________________________________________________________________________________________________________
ERROR:   py27: commands failed
```

As you can see it actually prints a coverage report to stdout, so I don't know why it thinks the test coverage is `None`.

Here's what's installed:

```
$ .tox/py27/bin/pip freeze | egrep 'pytest|cov'
You are using pip version 7.1.0, however version 7.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
cov-core==1.15.0
coverage==3.7.1
pytest==2.7.2
pytest-cov==2.0.0
pytest-xdist==1.13.1
```

**Comment by ionelmc** [comment]
Interesting. What's your command line and pytest options (specifically `addopts`)?

**Comment by ionelmc** [comment]
PS. `cov-core` is not needed anymore.

**Comment by msabramo** [comment]
It's looking like this *might* be an interaction between pytest-cov 2.0 and [teamcity-messages](https://pypi.python.org/pypi/teamcity-messages), because it fails when the `py.test` command-line includes `--teamcity` and it seems to be okay without it.

Hmmm... I'll post more as I find out more...

**Comment by msabramo** [comment]
Hmmm, upgrading from teamcity-messages 1.9.4 to 1.14 seems to fix the problem.

Hope this helps someone.

**Comment by yoloseem** [comment]
Well. I'm still getting this message on Travis CI with `pytest == 2.9.2`, `pytest-cov == 2.3.0` and `pytest-xdist == 1.14`. However there was no problem to get a coverage result with CodeCov even it printed erroneous output.

Full messages were like:
```bash
(...)
[gw1] PASSED tests/web/util_test.py::test_slc 
[gw0] PASSED tests/web/util_test.py::test_slc_entry INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`

---------- coverage: platform linux2, python 2.7.9-final-0 -----------
Coverage XML written to file coverage.xml

=================== 1062 passed, 1 skipped in 313.52 seconds ===================
The command "py.test -n2 --durations=20 --database-url 'postgresql:///travis_ci_test?user=postgres' --cov-config .coveragerc --cov-report=xml --cov=beat -vv -x --rerun 5 tests/" exited with 0.
```

**Comment by ionelmc** [comment]
@yoloseem what's your `coverage` version?

**Comment by yoloseem** [comment]
@ionelmc Didn't pinned explicitly but `4.1` is installed by `coverage>=3.7.1 (from pytest-cov>=2.3.0`). Tried again with the latest 4.2b1 but the problem remains.

**Comment by yhager** [comment]
It only happens to me when I specify `-n` in py.test invocation. Even `-n 1` causes it.
```
platform linux -- Python 3.5.2, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
plugins: cov-2.3.0, xdist-1.14, capturelog-0.7
```

**Comment by ionelmc** [comment]
@yhager I'm pretty sure your problem is #124 which is fixed in 2.3.1

**Comment by yhager** [comment]
@ionelmc that's correct. I upgraded to 2.3.1 and I can't recreate it. Thx.
